doc/devel/testing.rst

```diff
@@ -20,6 +21,13 @@ infrastructure are in :mod:`matplotlib.testing`.
 Requirements
 ------------
 
+We strongly recommend developing new features in a seperate virtualenv.
+
+Install all the Matplotlib dependencies_, and install matplotlib into the
+virtualenv with ``pip install -e matplotlib`` (but note the changes needed to
+``setup.cfg`` below) which will allow you to edit the code without
+reinstalling.
```

Comment:
I still think this is repeating what's in `devel/contributing.rst`.  I think we should avoid duplicating instructions. If you haven't already installed the development version of Matplotlib then testing doesn't make a lot of sense.

Reply:
I would just write "Install the latest version of Matplotlib as documented in http://matplotlib.org/devdocs/devel/contributing.html#retrieving-and-installing-the-latest-version-of-the-code.  In particular, follow the instructions to use a local FreeType build." (fix the link, of course).  And then remove the mention of localfreetype below.

(We also print a warning when tests are done on a nonlocal freetype version so I think that's enough.)